src/snowflake/connector/auth/keypair.py

```diff
@@ -63,22 +68,27 @@ def __init__(self, private_key, lifetime_in_seconds: int = LIFETIME):
                 "JWT_CNXN_RETRY_ATTEMPTS", AuthByKeyPair.DEFAULT_JWT_RETRY_ATTEMPTS
             )
         )
-        self._jwt_cnxn_wait_time = timedelta(
+        self._timeout = timedelta(
             seconds=int(
                 os.getenv(
                     "JWT_CNXN_WAIT_TIME", AuthByKeyPair.DEFAULT_JWT_CNXN_WAIT_TIME
                 )
             )
-        )
+        ).seconds
         self._current_retry_count = 0
 
-    def authenticate(
+    def reset_secrets(self) -> None:
```

Comment:
should we also reset `self._jwt_token` here?

Reply:
JWT tokens are safe to share to anyone, see https://medium.com/jspoint/so-what-the-heck-is-jwt-or-json-web-token-dca8bcb719a6